contracts/token-vesting.clar

```diff
@@ -5,12 +5,48 @@
 ;; Withdraws are unlocked to a specified set of addresses
 ;; at a certain schedule and conditions.
 
-;; Treasury map. Stores a "depositor" => "vesting" relation.
+;; SIP-010 token trait.
+(use-trait ft-trait .sip-010-trait.sip-010-trait)
+
+;;
+;; -- Data
+;;
+
+;; Treasury map.
+;; Identify and stores vestings for the contract.
 (define-map treasury
-  principal {token: (optional principal), amount: uint})
+  uint {sender: principal, amount: uint, token-contract: principal})
+
+;; Vesting ID.
+(define-data-var vesting-id uint u0)
+
+;; Vesting Vault.
+(define-data-var vesting-vault principal 'ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5)
```

Comment:
Q: why did you define this as a variable? It's probably more reasonable that the smart contract itself acts as the vault to keep the unvested tokens.

Smart contracts in Clarity don't have private keys associated with them. This means by default that only functions defined on that smart contract can manipulate tokens stored in it. This is a lot safer, since you specify an unknown address here where users have to trust who has the private keys.

Reply:
Was hardcoded only for making this initial implementation useful.

And thanks for the clear explanation. I'll change that. 👍